src/pelib/RichHeader.cpp

```diff
@@ -13,6 +13,686 @@
+		"Utc1700_C",
+		"Utc1700_CPP",
+		"Utc1700_CVTCIL_C",
+		"Utc1700_CVTCIL_CPP",
+		"Utc1700_LTCG_C",
+		"Utc1700_LTCG_CPP",
+		"Utc1700_LTCG_MSIL",
+		"Utc1700_POGO_I_C",
+		"Utc1700_POGO_I_CPP",
+		"Utc1700_POGO_O_C",
+		"Utc1700_POGO_O_CPP",
+		"AliasObj1200",
+		"Cvtpgd1800",
+		"Cvtres1200",
+		"Export1200",
+		"Implib1200",
+		"Linker1200",
+		"Masm1200",
+		"Utc1800_C",
+		"Utc1800_CPP",
+		"Utc1800_CVTCIL_C",
+		"Utc1800_CVTCIL_CPP",
+		"Utc1800_LTCG_C",
+		"Utc1800_LTCG_CPP",
+		"Utc1800_LTCG_MSIL",
+		"Utc1800_POGO_I_C",
+		"Utc1800_POGO_I_CPP",
+		"Utc1800_POGO_O_C",
+		"Utc1800_POGO_O_CPP",
+		"AliasObj1210",
+		"Cvtpgd1810",
+		"Cvtres1210",
+		"Export1210",
+		"Implib1210",
+		"Linker1210",
+		"Masm1210",
+		"Utc1810_C",
+		"Utc1810_CPP",
+		"Utc1810_CVTCIL_C",
+		"Utc1810_CVTCIL_CPP",
+		"Utc1810_LTCG_C",
+		"Utc1810_LTCG_CPP",
+		"Utc1810_LTCG_MSIL",
+		"Utc1810_POGO_I_C",
+		"Utc1810_POGO_I_CPP",
+		"Utc1810_POGO_O_C",
+		"Utc1810_POGO_O_CPP",
+		"AliasObj1400",
+		"Cvtpgd1900",
+		"Cvtres1400",
+		"Export1400",
+		"Implib1400",
+		"Linker1400",
+		"Masm1400",
+		"Utc1900_C",
+		"Utc1900_CPP",
+		"Utc1900_CVTCIL_C",
+		"Utc1900_CVTCIL_CPP",
+		"Utc1900_LTCG_C",
+		"Utc1900_LTCG_CPP",
+		"Utc1900_LTCG_MSIL",
+		"Utc1900_POGO_I_C",
+		"Utc1900_POGO_I_CPP",
+		"Utc1900_POGO_O_C",
+		"Utc1900_POGO_O_CPP"
+	};
+
+	// Array of visualStudioNameIndex -> name of Visual Studio
+	const std::vector<std::string> visualStudioNames =
+	{
+		".NET Framework",               //  0
+		".NET Core",                    //  1
+		"Visual Studio 5.0 CvtRes.exe", //  2
+		"Visual C++ 5.0",               //  3
+		"Visual Studio 6.0",            //  4
+		"Visual Basic 6.0",             //  5
+		"Visual C++ 6.0",               //  6
+		"Visual Studio 6.0 MASM",       //  7
+		"Visual C++ 2002",              //  8
+		"Visual C++ 2003",              //  9
+		"Visual Studio 2005",           // 10
+		"Visual Studio 2008",           // 11
+		"Visual Studio 2010",           // 12
+		"Visual C++ 2012",              // 13
+		"Visual Studio 2012 MASM",      // 14
+		"Visual C++ 2013",              // 15
+		"Visual Studio 2013",           // 16
+		"Visual C++ 2015",              // 17
+		"Visual Studio 2015",           // 18
+		"Visual C++ 2017",              // 19
+		"Visual Studio 2017",           // 20
+		"Visual C++ 2019",              // 21
+		"Visual Studio 2019"            // 22
+	};
+
+	// Array of index -> Visual Studio Version
+	const std::vector<std::string> visualStudioNames2 =
+	{
+		"Visual Studio 2003 v7.10",
+		"Visual Studio 2005 v8.0",
+		"Visual Studio 2008 v9.0",
+		"",
+		"Visual Studio 2010 v10.0",
+		"Visual Studio 2010 v10.10",
+		"Visual Studio 2012 v11.0",
+		"Visual Studio 2013 v12.0",
+		"Visual Studio 2013 v12.10",
+		"Visual Studio 2015+"
+	};
+
+	// Key: Build number from RichHeader
+	// Value: Pair of { visualStudioNameIndex, visualStudioVersionString }
+	const std::map<size_t, std::pair<size_t, std::string>> visualStudioVersionMap
```

Comment:
Is ordering important for this lookup table? If not, I would stick with `std::unordered_map` for faster lookup.